stdlib/public/CMakeLists.txt

```diff
@@ -19,6 +19,19 @@ if(SWIFT_STDLIB_SIL_DEBUGGING)
   list(APPEND SWIFT_RUNTIME_SWIFT_COMPILE_FLAGS "-Xfrontend" "-sil-based-debuginfo")
 endif()
 
+option(SWIFT_RUNTIME_PART_OF_DARWIN_SHARED_CACHE
+       "On Darwin platforms, link the standard library in a way that
+       allows inclusion in the dyld shared cache"
+       FALSE)
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND
+   LLVM_LINKER_IS_APPLE AND
```

Comment:
`LLVM_LINKER_IS_APPLE` will be introduced with https://github.com/llvm/llvm-project/pull/77806.

Keeping this as a draft PR for now until I understand the outcome of the review of the LLVM one.